spark_log_parser/eventlog.py

```diff
@@ -26,41 +24,58 @@ def _validate_work_dir(self, work_dir: Path | str) -> Path:
 
         return work_dir_path
 
-    def build(self) -> Path:
+    def build(self) -> tuple[Path, bool]:
 
         if not self.event_log_paths:
             raise ValueError("No files found")
 
-        self.event_log = self._get_event_log(self.event_log_paths)
+        self.event_log, self.parsed = self._get_event_log(self.event_log_paths)
 
-        return self.event_log
+        return self.event_log, self.parsed
+
+    def _get_event_log(self, paths: list[Path]) -> tuple[Path, bool]:
 
-    def _get_event_log(self, paths: list[Path]) -> Path:
         log_files = []
         rollover_dat = []
+        parsed = False
         for path in paths:
-            with open(path) as fobj:
-                try:
+            try:  # Test if it is a raw log
+                with open(path) as fobj:
                     line = json.loads(fobj.readline())
+                    if "Event" in line:
+                        log_files.append(path)
+                        if line["Event"] == "DBCEventLoggingListenerMetadata":
+                            rollover_dat.append(
+                                (line["Rollover Number"], line["SparkContext Id"], path)
+                            )
+                    else:
+                        raise ValueError
+
+            except ValueError:
+                try:  # Test if it is a parsed log
+                    with open(path) as fobj:
+                        data = json.load(fobj)
+                        if "jobData" in data:
```

Comment:
Ideally, `sparkApplication` should be making the determination whether the object is a valid event log or not. It creates the parsed logs after all.

Reply:
In principle I agree, and in the future I we can make that happen. For now though, you'll see in the sync_backend sister PR that parsed/unparsed judgement has to come before the log has been parsed because that must be known before entering the `spark_log_parser` through `SparkApplicationAdvanced`, which inherits from `sparkApplication`. I know, it's real janky right now, but it's getting better bit by bit. A separation of these classes will happen, and we can make this even cleaner then -- I don't want to get carried away in the one PR.

Reply:
Just to be clear, I was thinking it could be a `classmethod` like,
```
T = TypeVar('T', bound='sparkApplication')


class sparkApplication:
    @classmethod
    def is_parsed_log(cls: Type[T], eventlog: dict) -> boolean:
        ...
```
(btw, I learned the type annotations for classmethods here: https://stackoverflow.com/questions/44640479/type-annotation-for-classmethod-returning-instance)
which allows us to call the method before we create an instance of the class. That way we keep the parsed log logic in the same module.

I totally agree that we are making good steady progress. It's a balance between improving the structure, not losing any of the knowledge in the code, and delivering features n' fixes.

Reply:
...or `staticmethod` if we don't need that class variable

Reply:
ohh I see what you mean, that's nice. I'll keep this in mind for the next batch of log_parser work.